src/FloatingDockContainer.cpp

```diff
@@ -1035,6 +1039,18 @@ QList<CDockWidget*> CFloatingDockContainer::dockWidgets() const
 	return d->DockContainer->dockWidgets();
 }
 
+//============================================================================
+void CFloatingDockContainer::hideAndDeleteLater()
+{
+	// Widget has been redocked, so it must be hidden right way (see 
+	// https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/issues/351)
+	// but AutoHideChildren must be set to false because "this" still contains
+	// dock widgets that shall not be toggled hidden.
+	d->AutoHideChildren = false;

```

Comment:
Where is `AutoHideChildren` reset to `true` again if it is set to `false` here? I think it should be reset to `true` in the `hideEvent` function.

Reply:
We did not consider it necessary to reset `AutoHideChildren` to `true` because the floating dock widget is planned to be destroyed anyway. Moreover if the floating dock widget is hidden another time before its destruction, then it allows to be 100% sure the bug is fixed (else, if two `QHideEvent` are received in a row, bug will be fixed upion first call, but will reappear upon the second one). But if you prefer there is no problem to reset `AutoHideChildren` to `true`.

Reply:
Ok, sorry, this was my fault - you are right, the dock widget will be destroyed anyway.